pkg/action/release_testing.go

```diff
@@ -43,52 +42,16 @@ func NewReleaseTesting(cfg *Configuration) *ReleaseTesting {
 }
 
 // Run executes 'helm test' against the given release.
-func (r *ReleaseTesting) Run(name string) (<-chan *release.TestReleaseResponse, <-chan error) {
-	errc := make(chan error, 1)
+func (r *ReleaseTesting) Run(name string) error {
 	if err := validateReleaseName(name); err != nil {
-		errc <- errors.Errorf("releaseTest: Release name is invalid: %s", name)
-		return nil, errc
+		return errors.Errorf("releaseTest: Release name is invalid: %s", name)
 	}
 
 	// finds the non-deleted release with the given name
 	rel, err := r.cfg.Releases.Last(name)
 	if err != nil {
-		errc <- err
-		return nil, errc
+		return err
 	}
 
-	ch := make(chan *release.TestReleaseResponse, 1)
-	testEnv := &reltesting.Environment{
-		Namespace:  rel.Namespace,
-		KubeClient: r.cfg.KubeClient,
-		Timeout:    r.Timeout,
-		Messages:   ch,
-	}
-	r.cfg.Log("running tests for release %s", rel.Name)
-	tSuite := reltesting.NewTestSuite(rel)
-
-	go func() {
-		defer close(errc)
-		defer close(ch)
-
-		if err := tSuite.Run(testEnv); err != nil {
-			errc <- errors.Wrapf(err, "error running test suite for %s", rel.Name)
-			return
-		}
-
-		rel.Info.LastTestSuiteRun = &release.TestSuite{
```

Comment:
Could you talk us through why you want to delete this part? Do you plan on re-implementing?

Reply:
Replaced with generic hook execution tracking in 593b6a04b9e88ca490816991cb7e2723c63ef088.